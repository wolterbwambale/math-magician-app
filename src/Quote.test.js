import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Quote from './components/quote';

describe('Quote component', () => {
  test('renders loading state initially', () => {
    const { container } = render(<Quote />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders error state if fetching quotes fails', async () => {
    jest.spyOn(window, 'fetch').mockImplementationOnce(() => {
      throw new Error('Fetch error');
    });

    const { container } = render(<Quote />);
    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  test('renders quotes when fetched successfully', async () => {
    const mockQuotes = [
      {
        author: 'John Doe',
        quote: 'Sample quote',
      },
    ];

    jest.spyOn(window, 'fetch').mockImplementationOnce(() => Promise.resolve({
      json: () => Promise.resolve(mockQuotes),
    }));

    const { container } = render(<Quote />);
    await act(async () => {
      await waitFor(() => {
        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
