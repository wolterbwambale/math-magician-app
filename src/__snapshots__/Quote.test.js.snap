// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders error state if fetching quotes fails 1`] = `
<div>
  Something went wrong!
</div>
`;

exports[`Quote component renders loading state initially 1`] = `
<div>
  Loading...
</div>
`;

exports[`Quote component renders quotes when fetched successfully 1`] = `
<div>
  Loading...
</div>
`;
